include/hip/hcc_detail/hip_runtime.h

```diff
@@ -305,11 +310,15 @@ static constexpr Coordinates<hip_impl::WorkitemId> threadIdx{};
 
 #endif // defined __HCC__
 #if __HCC_OR_HIP_CLANG__
+#if __HIP_ENABLE_MALLOC__
 extern "C" __device__ void* __hip_malloc(size_t);
 extern "C" __device__ void* __hip_free(void* ptr);
-
 static inline __device__ void* malloc(size_t size) { return __hip_malloc(size); }
 static inline __device__ void* free(void* ptr) { return __hip_free(ptr); }
+#else
+static inline __device__ void* malloc(size_t size) { return nullptr; }
+static inline __device__ void* free(void* ptr) { return nullptr; }
```

Comment:
Perhaps having a compilation error would be better?

Reply:
There may be cases they need these functions to compile.
They will fail at runtime anyway.

Reply:
It's much easier to fix a build error than to debug a runtime failure.

Reply:
ok. removed